### PR TITLE
Convert packages/jaeger-ui-components/src/TracePageHeader/TracePageSearchBar.test.js to RTL

### PR DESCRIPTION
**Pull request by joey-grafana**

### CODE REVIEW

File: packages/jaeger-ui-components/src/common/UiFindInput.tsx [hunk 1]
```diff
@@ -56,6 +57,7 @@ export default class UiFindInput extends React.PureComponent<Props> {
         onChange={(e) => this.props.onChange(e.currentTarget.value)}
         suffix={suffix}
         value={value}
+        data-testid={selectors.components.TraceViewer.uiFindInput}
```
Comment (ashharrison90) [inline packages/jaeger-ui-components/src/common/UiFindInput.tsx:60]: shouldn't need a testid here

File: packages/jaeger-ui-components/src/TracePageHeader/TracePageSearchBar.tsx [hunk 2]
```diff
@@ -184,7 +187,7 @@ export default memo(function TracePageSearchBar(props: TracePageSearchBarProps)
                 disabled={!searchValue}
                 type="button"
                 icon="arrow-up"
-                data-testid="trace-page-search-bar-prev-result-button"
+                data-testid={selectors.components.TraceViewer.tracePageSearchBarPrevResultButton}
```
Comment (ashharrison90) [inline packages/jaeger-ui-components/src/TracePageHeader/TracePageSearchBar.tsx:190]: shouldn't need a testid here

File: packages/jaeger-ui-components/src/TracePageHeader/TracePageSearchBar.tsx [hunk 3]
```diff
@@ -175,7 +178,7 @@ export default memo(function TracePageSearchBar(props: TracePageSearchBarProps)
                 disabled={!searchValue}
                 type="button"
                 icon="arrow-down"
-                data-testid="trace-page-search-bar-next-result-button"
+                data-testid={selectors.components.TraceViewer.tracePageSearchBarNextResultButton}
```
Comment (ashharrison90) [inline packages/jaeger-ui-components/src/TracePageHeader/TracePageSearchBar.tsx:181]: shouldn't need a testid here

File: packages/jaeger-ui-components/src/TracePageHeader/TracePageSearchBar.tsx [hunk 4]
```diff
@@ -95,14 +95,17 @@ export default memo(function TracePageSearchBar(props: TracePageSearchBarProps)
...
+      data-testid={selectors.components.TraceViewer.tracePageSearchBarSuffix}
+    >
       {searchBarSuffix}
     </span>
   ) : null;
 
   const btnClass = cx(styles.TracePageSearchBarBtn, { [styles.TracePageSearchBarBtnDisabled]: !searchValue });
   const uiFindInputInputProps = {
-    'data-test': markers.IN_TRACE_SEARCH,
+    'data-test': selectors.components.TraceViewer.tracePageSearchBar,
```
Comment (ashharrison90) [inline packages/jaeger-ui-components/src/TracePageHeader/TracePageSearchBar.tsx:108]: don't quite understand what this is doing/why we need this? 🤔

File: packages/grafana-e2e-selectors/src/selectors/components.ts [hunk 5]
```diff
@@ -266,6 +266,11 @@ export const Components = {
   },
   TraceViewer: {
     spanBar: 'data-testid SpanBar--wrapper',
+    tracePageSearchBar: 'data-testid-trace-page-search-bar',
+    tracePageSearchBarNextResultButton: 'data-testid-trace-page-search-bar-next-result-button',
+    tracePageSearchBarPrevResultButton: 'data-testid-trace-page-search-bar-prev-result-button',
+    tracePageSearchBarSuffix: 'data-testid-trace-page-search-bar-suffix',
+    uiFindInput: 'data-testid-ui-find-input',
```
Comment (ashharrison90) [inline packages/grafana-e2e-selectors/src/selectors/components.ts:273]: these should be in the style `data-testid <id>`

**Review by ashharrison90 — CHANGES_REQUESTED** [review]
great start! 🙌 

the main thing we should change is to focus on using accessible selectors instead of relying on `data-testid` too much. `data-testid` is an escape hatch we should only use when there is no other way of selecting the element. `tracePageSearchBar` looks like a good example: it's a containing `div` with no other real means of selecting it. 

all the other elements look like they can be selected either by role + name (preferred) or by text. see https://testing-library.com/docs/queries/about/#priority.

for example, `tracePageSearchBarPrevResultButton` and `tracePageSearchBarNextResultButton`. you should be able to get these using `screen.getByRole('button', { name: <something> })`. if there is no name then that's a problem for any user with a screen reader and we should fix it by adding an appropriate `ariaLabel` to the `<Button>`. 

hope that makes sense! 😄

File: packages/jaeger-ui-components/src/TracePageHeader/TracePageSearchBar.tsx (hunk 4 above)
Comment (joey-grafana) [reply packages/jaeger-ui-components/src/TracePageHeader/TracePageSearchBar.tsx:108]: Removed!

**Review by ashharrison90 — APPROVED** [review]
sorry for the late rereview, been off with national holidays 😄 

this looks great! 🥳

**Pull request merged**